crates/llm-ls/src/adaptors.rs

```diff
@@ -0,0 +1,142 @@
+use super::{
+    internal_error, APIError, APIResponse, CompletionParams, Generation, Ide, NAME, VERSION,
+};
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION, USER_AGENT};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tower_lsp::jsonrpc;
+
+struct AdaptHuggingFaceRequest;
+impl AdaptHuggingFaceRequest {
+    fn adapt_body(&self, prompt: String, params: CompletionParams) -> Value {
+        return serde_json::json!({
+            "inputs": prompt,
+            "parameters": params.request_params,
+        });
+    }
+    fn adapt_headers(
+        &self,
+        api_token: Option<&String>,
+        ide: Ide,
+    ) -> Result<HeaderMap, jsonrpc::Error> {
+        let mut headers = HeaderMap::new();
+        let user_agent = format!("{NAME}/{VERSION}; rust/unknown; ide/{ide:?}");
+        headers.insert(
+            USER_AGENT,
+            HeaderValue::from_str(&user_agent).map_err(internal_error)?,
+        );
+
+        if let Some(api_token) = api_token {
+            headers.insert(
+                AUTHORIZATION,
+                HeaderValue::from_str(&format!("Bearer {api_token}")).map_err(internal_error)?,
+            );
+        }
+
+        Ok(headers)
+    }
+}
+
+struct AdaptHuggingFaceResponse;
+impl AdaptHuggingFaceResponse {
+    fn adapt_blob(&self, text: reqwest::Result<String>) -> Result<Vec<Generation>, jsonrpc::Error> {
+        let generations =
+            match serde_json::from_str(&text.unwrap_or_default()).map_err(internal_error)? {
+                APIResponse::Generation(gen) => vec![gen],
+                APIResponse::Generations(gens) => gens,
+                APIResponse::Error(err) => return Err(internal_error(err)),
+            };
+        Ok(generations)
+    }
+}
+
+struct AdaptOllamaRequest;
```

Comment:
```suggestion
struct OllamaRequest;
```

The `Adapt` prefix seems redundant